gpbft/emulator/emulator.go

```diff
@@ -0,0 +1,100 @@
+package emulator
+
+import (
+	"github.com/filecoin-project/go-bitfield"
+	"github.com/filecoin-project/go-f3/gpbft"
+	"slices"
+)
+
+// Emulator emulates the core GPBFT protocol for one honest participant, and
+// allows frame-by-frame control over progress of a GPBFT instance.
+//
+// Progress may be made by calling Emulator.DeliverMessage,
+// Emulator.ReceiveBroadcast or Emulator.DeliverAlarm.
+type Driver struct {
+	subject *gpbft.Participant
+	host    *mockHost
+	require *Assertions
+}
+
+// New instantiates a new Emulator with the given GPBFT options. See
+// Emulator.Start.
+func New(assertions *Assertions, o ...gpbft.Option) *Driver {
```

Comment:
May be a YAGNI but the reason i did not do this was to avoid having emulator package depend on `testing`. This is so that the emulator can be used at production level code in the future for non-testing scenarios: for example using emulator as a light-weight tool to research core protocol improvements. That would separate the emulated behaviour from what it considers "correct". 

If YAGNI, then I would take `testing.T` here straight up.

Reply:
I think I would use the simulator for protocol R&D (that's what it was first built for). This is for unit tests.

Reply:
You are probably right, thanks